db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -3314,7 +3314,8 @@ Status DBImpl::BackgroundCompaction(bool* made_progress,
       if (start_level > 0) {
         auto vstorage = c->input_version()->storage_info();
         c->edit()->AddCompactCursor(
-            start_level, vstorage->GetNextCompactCursor(start_level));
+            start_level,
+            vstorage->GetNextCompactCursor(start_level, c->num_input_files(0)));
```

Comment:
[JustMeLearning] Why did we hardcoded 0 in c->num_input_files(0) here?

Reply:
https://github.com/facebook/rocksdb/blob/66685d6aa184ec3905428fb5255cbd6ec20bf242/db/compaction/compaction.h#L114

That returns the number of input files from the upper level in a compaction. Since we compact that many number of files, the cursor should be advanced by so many positions.

Reply:
0 here refers the starting level in one compaction. Say one compaction from level 2 to 3, 0 here refers to the level 2.